drivers/gpio/sabertooth_analog_driver.go

```diff
@@ -0,0 +1,294 @@
+
+// NewSabertoothDIPSwitches creates a new SabertoothDIPSwitches given the
+// position (true: ON, false: OFF) of the DIP switches 3, 4, 5 and 6.
+func NewSabertoothDIPSwitches(powerSupply, mixed, linear, bidirectional bool) SabertoothDIPSwitches {
+    // Transforms booleans into uint8
+    i := func(b bool) SabertoothDIPSwitches {
+        if b { return 1 } else { return 0 }
+    }
+    
+    // The first two DIP switches sets the driver in analog mode
+    return 0b0000_0011 |
+        i(powerSupply) << 2 |
+        i(mixed) << 3 |
+        i(linear) << 4 |
+        i(bidirectional) << 5
+}
+
+// DIPSwitch1 returns true if the DIP switch 1 is ON
+func (d SabertoothDIPSwitches) DIPSwitch1() bool { return (0b0000_0001 & d) > 0 }
+// DIPSwitch2 returns true if the DIP switch 2 is ON
+func (d SabertoothDIPSwitches) DIPSwitch2() bool { return (0b0000_0010 & d) > 0 }
+// DIPSwitch3 returns true if the DIP switch 3 is ON
+func (d SabertoothDIPSwitches) DIPSwitch3() bool { return (0b0000_0100 & d) > 0 }
+// DIPSwitch4 returns true if the DIP switch 4 is ON
+func (d SabertoothDIPSwitches) DIPSwitch4() bool { return (0b0000_1000 & d) > 0 }
+// DIPSwitch5 returns true if the DIP switch 5 is ON
+func (d SabertoothDIPSwitches) DIPSwitch5() bool { return (0b0001_0000 & d) > 0 }
+// DIPSwitch6 returns true if the DIP switch 6 is ON
+func (d SabertoothDIPSwitches) DIPSwitch6() bool { return (0b0010_0000 & d) > 0 }
+
+// PowerSupply returns true if the DIP switch 3 is ON
+func (d SabertoothDIPSwitches) PowerSupply() bool { return d.DIPSwitch3() }
+
+// Battery returns true if the DIP switch 3 is OFF
+func (d SabertoothDIPSwitches) Battery() bool { return !d.DIPSwitch3() }
+
+// Mixed returns true if the DIP switch 4 is ON
+func (d SabertoothDIPSwitches) Mixed() bool { return d.DIPSwitch4() }
+
+// Independent returns true if the DIP switch 4 is OFF
+func (d SabertoothDIPSwitches) Independent() bool { return !d.DIPSwitch4() }
+
+// Linear returns true if the DIP switch 5 is ON
+func (d SabertoothDIPSwitches) Linear() bool { return d.DIPSwitch5() }
+
+// Exponential returns true if the DIP switch 5 is OFF
+func (d SabertoothDIPSwitches) Exponential() bool { return !d.DIPSwitch5() }
+
+// Bidirectional returns true if the DIP switch 6 is ON
+func (d SabertoothDIPSwitches) Bidirectional() bool { return d.DIPSwitch6() }
+
+// Unidirectional returns true if the DIP switch 6 is OFF
+func (d SabertoothDIPSwitches) Unidirectional() bool { return !d.DIPSwitch6() }
+
+type sabertoothAnalogDriver struct {
+    name         string
+    switches     SabertoothDIPSwitches
+    s1           string
+    s2           string
+    a1           string
+    a2           string
+    throttle     float64
+    differential float64
+    
+    // Used for differential change
+    m1Ratio, m2Ratio float64
+    
+    connection PwmWriter
+    gobot.Commander
+}
+
+// SabertoothAnalogDriver represents a sabertooth driver in analog mode
+type SabertoothAnalogDriver interface {
```

Comment:
Please could you remove the interface and add the comments to the related exported functions.This is, because there is no further implementation of it and introducing of interfaces in golang is a little bit different according to other languages.

After that you could most likely also remove the "// END" comments in the code, because the beginning of the next function is marked by the description comment.

Reading [this post](https://www.integralist.co.uk/posts/go-interfaces/) will help to understand my remark.